corehq/change_ownership.py

```diff
@@ -0,0 +1,202 @@
+from corehq.apps.users.models import CommCareUser
+from corehq.form_processor.models import CommCareCase
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.es import CaseSearchES
+from corehq.apps.es.cases import case_type
+import os
+import time
+import math
+
+from casexml.apps.case.mock import CaseBlock
+from corehq.apps.hqcase.utils import submit_case_blocks
+
+# DOMAIN = 'alafiacomm'
+DOMAIN = 'alafiacomm-prod'
+
+success_case_log_file_path = os.path.expanduser('~/script_success.log')
+error_case_log_file_path = os.path.expanduser('~/script_error.log')
+
+BATCH_SIZE = 100
+SKIP_COUNT = 0
+
+def submit_cases(case_blocks):
+    submit_case_blocks(
+        [cb.as_text() for cb in case_blocks],
+        domain=DOMAIN,
+        device_id='system',
+    )
+
+def write_to_log(ids, is_success=True, message=None):
+    if is_success:
+        file_path = success_case_log_file_path
+    else:
+        file_path = error_case_log_file_path
+    with open(file_path, 'a') as log:
+        for id in ids:
+            log.write(
+                'Skipped ' if not is_success else '' \
+                f'{id} - ' \
+                'Reason:' + message if message else ''
+            )
+        log.close()
+
+def process_batch(case_blocks, current_chunk, total_count):
+    batch_start = current_chunk * BATCH_SIZE + 1
+    batch_end = batch_start + len(case_blocks)
+    percentage_done = round((batch_start / total_count) * 100, 2)
+    print(f'Submitting cases {batch_start}-{batch_end}/{total_count} ({percentage_done}%)')
+    try:
+        submit_cases(case_blocks)
+        write_to_log([cb.case_id for cb in case_blocks])
+        return True
+    except Exception as e:
+        write_to_log(
+            [cb.case_id for cb in case_blocks],
+            is_success=False,
+            message=str(e)
+        )
+        return False
+
+### Task 1
+def move_mobile_workers():
+    print("---MOVING MOBILE WORKER LOCATIONS---")
+    RC_NUM_PROP_NAME = 'rc_number'
+    USER_TYPE_PROP_NAME = 'usertype'
+
+    users = CommCareUser.by_domain(DOMAIN)
+    user_count = len(users)
+    success_count = fail_count = skip_count = 0
+    total_time = 0
+    print(f"Total Users to Process: {user_count}")
+    user_to_save = []
+    for idx, user in enumerate(users):
+        start_time = time.time()
+        percentage_done = round((idx / user_count) * 100, 2)
```

Comment:
you could use `with_progress_bar` to do this for you.

Reply:
I'll take a look, thanks!

Reply:
I've gotten rid of all the timing/percentage metrics, making use of `with_progress_bar` ([bf3b264](https://github.com/dimagi/commcare-hq/pull/34588/commits/bf3b264f740a226213cd2df2fbe86f6fd632fcb8))